src/main/kotlin/org/rust/ide/intentions/NestUseStatementsIntention.kt

```diff
@@ -0,0 +1,207 @@
+                val useItemOnCursor = useItem
+
+                val basePath = useSpeck.path?.basePath()?.text ?: return null
+                useItemList += useItemOnCursor.leftSiblings.filterIsInstance<RsUseItem>().filter {
+                    it.useSpeck?.path?.basePath()?.text?.startsWith(basePath) ?: false
+                }
+                useItemList.add(useItemOnCursor)
+                useItemList += useItemOnCursor.rightSiblings.filterIsInstance<RsUseItem>().filter {
+                    it.useSpeck?.path?.basePath()?.text?.startsWith(basePath) ?: false
+                }
+                if (useItemList.size == 1) return null
+                val firstUseItem = useItemList.first()
+
+                val useSpeckList = useItemList
+                    .mapNotNull { it.useSpeck }
+
+                return PathInUseItem(useItem, useSpeck, useSpeckList, firstUseItem, useItemList)
+            }
+        }
+
+        override fun createElement(path: String, project: Project): PsiElement {
+            return RsPsiFactory(project).createUseItem(path)
+        }
+
+        override fun getPreviousElements(): List<PsiElement> {
+            return useItems
+        }
+
+        override fun firstPreviousElement(): PsiElement {
+            return firstUseItem
+        }
+
+        override fun root(): PsiElement {
+            return useItem.parent
+        }
+    }
+
+    class PathInGroup(val useGroup: RsUseGroup, val useSpeck: RsUseSpeck, override val useSpecks: List<RsUseSpeck>,
+                      val firstUseSpeck: RsUseSpeck) : Context {
+        override val cursorOffset: Int = 0
+
+        companion object {
+            fun create(useGroup: RsUseGroup, useSpeck: RsUseSpeck): PathInGroup? {
+                val useSpeckList = mutableListOf<RsUseSpeck>()
+                val useSpeckOnCursor = useSpeck
+
+                val basePath = useSpeck.path?.basePath()?.text ?: return null
+                useSpeckList += useSpeckOnCursor.leftSiblings.filterIsInstance<RsUseSpeck>().filter {
+                    it.path?.basePath()?.text?.startsWith(basePath) ?: false
+                }
+                useSpeckList.add(useSpeckOnCursor)
+                useSpeckList += useSpeckOnCursor.rightSiblings.filterIsInstance<RsUseSpeck>().filter {
+                    it.path?.basePath()?.text?.startsWith(basePath) ?: false
+                }
+                if (useSpeckList.size == 1) return null
+                val firstUseSpeck = useSpeckList.first()
+
+                return PathInGroup(useGroup, useSpeck, useSpeckList, firstUseSpeck)
+            }
+        }
+
+        override fun createElement(path: String, project: Project): PsiElement {
+            return RsPsiFactory(project).createUseSpeck(path)
+        }
+
+        override fun getPreviousElements(): List<PsiElement> {
+            return useSpecks.flatMap {
+                val nextComma = if (it.nextSibling.text == ",") {
```

Comment:
It's better to compare element types instead of text:
`it.nextSibling.elementType == RsElementTypes.COMMA`